test/common/http/conn_manager_impl_test.cc

```diff
@@ -5070,9 +5070,8 @@ TEST_F(HttpConnectionManagerImplTest, FilterContinueAndEndStreamHeaders) {
   Buffer::OwnedImpl fake_input("1234");
   conn_manager_->onData(fake_input, true);
 
-  EXPECT_CALL(*encoder_filters_[1], encodeHeaders(_, true))
+  EXPECT_CALL(*encoder_filters_[1], encodeHeaders(_, false))
```

Comment:
The ASSERTs I added to make sure `ContinueAndEndStream` and `ContinueDontEndStream` are used correctly by filters failed on this test. After reading it, it seems like this was a bug. It makes since that `end_stream` is initially `false`, then the filter changes it to `true` by returning `ContinueAndEndStream`.

The updated version is also consistent with the other versions of the test,  `FilterContinueAndEndStreamData` and `FilterContinueAndEndStreamTrailers`. Which makes me believe more that this was a bug.

Reply:
Yes agreed this looks like a bug.